.github/workflows/core-security.yml

```
@@ -0,0 +1,56 @@
+        continue-on-error: true
+        env:
+          NODE_OPTIONS: "--max-old-space-size=4096"
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v3
+
+      - name: Check for secrets in code
+        run: |
+          # Simple grep for common secret patterns
+          ! grep -r -E "(API_KEY|api_key|apikey|api-key|password|secret|token)=\"[A-Za-z0-9/\+]{8,}\"" --include="*.{js,ts,tsx,jsx}" src/
+          
```

YAMLlint (1.35.1)

[error] trailing spaces

(trailing-spaces)